### PR TITLE
Create method for SubjectsController

### PR DESCRIPTION
**Pull request by VeronikaRevjakina**
#358

- [x] Add create method for SubjectsController

- [x] Add checkUser method for SubjectsController

- [x] testCreateSuccess 

- [x] testCreateWithIdFailed

- [x] testCheckUserBadRequest

- [x] testCreateForbidden

### CODE REVIEW

**Comment by VeronikaRevjakina** [comment]
@malinink please, check if create method is correct

**Comment by VeronikaRevjakina** [comment]
@malinink Can you please check the second mapper and give some advice how to fix it ? Thanks.

**Comment by VeronikaRevjakina** [comment]
@malinink I don’t know how to solve this CS error , there is no useless white spaces.

File: src/main/java/easytests/api/v1/controllers/SubjectsController.java [hunk 1]
```diff
@@ -58,9 +62,37 @@
...
+        return this.subjectsMapper.map(subjectModel, Identity.class);
+
+    }
+
+    private void checkUser(Subject subject) throws BadRequestException, ForbiddenException {
+        if (subject.getUser() != null) {
+            final UserModelInterface userModel = this.usersService.find(subject.getUser().getId());
+
+            if (!this.acl.hasAccess(userModel)) {
+                throw new ForbiddenException();
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/SubjectsController.java]: BadRequest

File: src/main/java/easytests/api/v1/controllers/SubjectsController.java [hunk 2]
```diff
@@ -58,9 +62,37 @@
...
+        subjectModel.setUser(usersService.find(subject.getUser().getId()));
+
+        this.subjectsService.save(subjectModel);
+
+        return this.subjectsMapper.map(subjectModel, Identity.class);
+
+    }
+
+    private void checkUser(Subject subject) throws BadRequestException, ForbiddenException {
+        if (subject.getUser() != null) {
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/SubjectsController.java]: check spec

File: src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java [hunk 3]
```diff
@@ -132,9 +139,83 @@ public void testListForbidden() throws Exception {
...
+        when(this.usersService.find(any(Integer.class))).thenReturn(userModel);
+        when(this.acl.hasAccess(any(UserModelInterface.class))).thenReturn(true);
+        final ArgumentCaptor<SubjectModelInterface> subjectCaptor = ArgumentCaptor.forClass(SubjectModelInterface.class);
+
+        mvc.perform(post("/v1/subjects")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new JsonSupport()
+                        .with(name, "Subject")
+                        .with(description, "Subject description")
+                        .with(user, new JsonSupport().with(id,2))
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java]: select different userId

File: src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java [hunk 4]
```diff
@@ -132,9 +139,83 @@ public void testListForbidden() throws Exception {
...
+    @Test
+    public void testCheckUserBadRequest() throws Exception {
+        Subject subject = new Subject();
+        mvc.perform(post("/v1/subjects").param("subject", "subject"))
+                .andExpect(status().isBadRequest());
+
+    }
+
+    @Test
+    public void testCreateForbidden() throws Exception {
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java]: badrequest

File: src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java [hunk 5]
```diff
@@ -132,9 +139,83 @@ public void testListForbidden() throws Exception {
...
+                        .with(user, new JsonSupport().with(2))
+                        .build()
+                ))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().string(""))
+                .andReturn();
+    }
+
+    @Test
+    public void testCheckUserBadRequest() throws Exception {
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/SubjectsControllerTest.java]: remove

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/java/easytests/api/v1/controllers/SubjectsController.java [hunk 6]
```diff
@@ -58,9 +62,34 @@
...
+    @ResponseStatus(HttpStatus.CREATED)
+    public Identity create(@RequestBody Subject subject) throws BadRequestException, ForbiddenException {
+        if (subject.getId() != null) {
+            throw new IdentifiedModelException();
+        }
+
+        this.checkUser(subject);
+
+        final SubjectModelInterface subjectModel = this.subjectsMapper.map(subject, SubjectModel.class);
+        subjectModel.setUser(usersService.find(subject.getUser().getId()));
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/SubjectsController.java]: forAuth

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by VeronikaRevjakina** [comment]
@malinink Coverage fails not because of my files :acl and userscontroller. Check,please.

**Review by malinink — APPROVED** [review]

**Pull request merged**